### PR TITLE
[WIP] DT Experiments

### PR DESCRIPTION
**Pull request by anirrudh**
This PR Includes a notebook alongside a working environment. The notebook provides a quick way to look at the current implementation of non-nanosecond date times within pandas. 

The `environment.yaml` should work to go and fetch the last hash from the pandas package as defined. 

This PR will need a rebase, followed by some squashes, but I've opened this PR for visibility, and it will not be ready to merge as long as it is marked WIP.

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
> pandas powered by NumPy

this is typically the case :) Perhaps "Out-of-bounds datetimes going via numpy" would be a better title?

I'm fairly confident that it'll be the same thing later, when this can happen directly within `to_datetime` or `read_csv`, it'll just be hidden to users

File: ENVIRONMENT.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Environment Setup
```
Comment (kcpevey) [inline ENVIRONMENT.md:1]: I'd like to see these changes added to the README

File: examples/L0_loading_to_series.ipynb [hunk 2]
```diff
@@ -91,6 +91,7 @@
    "execution_count": null,
    "id": "d5288b8e",
    "metadata": {
+    "collapsed": false,
```
Comment (kcpevey) [inline examples/L0_loading_to_series.ipynb:94]: the untouched notebooks don't need to be added here.

File: experiments/paleoclimdt/logic.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import pathlib
```
Comment (kcpevey) [inline experiments/paleoclimdt/logic.py:1]: this file is not necessary

File: experiments/time_exploration_pandas.ipynb [hunk 4]
```diff
@@ -0,0 +1,473 @@
+{
```
Comment (kcpevey) [inline experiments/time_exploration_pandas.ipynb:1]: This file can be added as a notebook in the examples folder that already exists. Please be sure to restart clear all

**Review by kcpevey — CHANGES_REQUESTED** [review]

**Comment by anirrudh** [comment]
Closing due to redundancies.

**Pull request closed without merging**